src/lib/ZoomControls.js

```diff
@@ -0,0 +1,140 @@
+import isFinite from 'lodash/isFinite';
+import EventEmitter from 'events';
+import { ICON_ZOOM_IN, ICON_ZOOM_OUT } from './icons/icons';
+import Controls from './Controls';
+import { ZOOM_CONTROLS_EVENTS } from './events';
+
+const CLASS_ZOOM_CURRENT_SCALE = 'bp-zoom-current-scale';
+const CLASS_ZOOM_IN_BUTTON = 'bp-zoom-in-btn';
+const CLASS_ZOOM_OUT_BUTTON = 'bp-zoom-out-btn';
+
+class ZoomControls extends EventEmitter {
+    /** @property {Controls} - Controls object */
+    controls;
+
+    /** @property {HTMLElement} - Controls element */
+    controlsElement;
+
+    /** @property {number} - Current zoom scale */
+    currentScale;
+
+    /** @property {HTMLElement} - Current scale element */
+    currentScaleElement;
+
+    /** @property {number} - Max zoom scale */
+    maxZoom;
+
+    /** @property {number} - Min zoom scale */
+    minZoom;
+
+    /**
+     * [constructor]
+     *
+     * @param {Controls} controls - Viewer controls
+     * @return {ZoomControls} Instance of ZoomControls
+     */
+    constructor(controls) {
+        super();
+
+        if (!controls || !(controls instanceof Controls)) {
+            throw Error('controls must be an instance of Controls');
+        }
+
+        this.controls = controls;
+        this.controlsElement = controls.controlsEl;
+
+        this.handleZoomIn = this.handleZoomIn.bind(this);
+        this.handleZoomOut = this.handleZoomOut.bind(this);
+    }
+
+    /**
+     * Add the zoom controls
+     *
+     * @param {number} currentScale - Initial scale value, assumes range of 0-1
+     * @param {number} [options.maxZoom] - Maximum zoom, range 0-1
+     * @param {number} [options.minZoom] - Minimum zoom, range 0-1
+     * @param {String} [options.zoomInClassName] - Class name for zoom in button
+     * @param {String} [options.zoomOutClassName] - Class name for zoom out button
+     * @return {void}
+     */
+    add(
+        currentScale,
+        { zoomOutClassName = '', zoomInClassName = '', minZoom = 0, maxZoom = Number.POSITIVE_INFINITY } = {},
```

Comment:
`maxZoom = Number.POSITIVE_INFINITY` what is the range of maxZoom? Is it 0-1 as the JSDoc says?

Reply:
`maxZoom` range is up to the consuming code, the scale is 0-1 though the maxZoom could be a value upwards of 1 -- I'll try to reflect this in the doc